### DISCUSSION TITLE
Element modification for LLM ASAG project

### DISCUSSION THREAD

**Discussion by Victorsss-Orz** [PrairieLearn/PrairieLearn#10618] [Ideas]
Hi all! This is Victor, the graduate student working on utilizing LLMs to perform Automatic Short Answer Grading. In the MVP of LLM ASAG, I'm rendering the question panel and submission panel and sending everything to the LLM. However, there can be a lot of redundant html code from the rendered panels and they are sent to the LLM as well. 

To improve grading accuracy by eliminating some irrelevant text, we are thinking about ways to simplify the html that gets rendered. @SethPoulsen and I have two proposals that might work: 
1. For the short term, create a custom text input element that performs minimum rendering (only the text input box)
2. On the long term, use a different render function for all elements (`renderForBotGrading` maybe) that only gets called by the LLM ASAG function with minimum html rendering

We wonder if these are valid things to do, or if there are things we need to be aware of before starting to work on these. Any feedback is appreciated!

**Comment by nwalters512** [comment]
Our recommendation would be to take the first option (custom course-specific element) as far as you can. This project is still way too early in the experimental research phase to make fundamental changes to PrairieLearn (like adding a new element phase) if we can avoid it.
